### PR TITLE
Create Sentry factory

### LINKED ISSUE #9031: Improve readability of our Sentry code
### Describe the problem that you'd like solved

Recently, I needed a way to expose the `Sentry` object linked to `ol-web`.  I didn't realize that it was already exposed [here](https://github.com/internetarchive/openlibrary/blob/e52cda783e48cec3ed5b14600a9b27dc89bdcbe6/openlibrary/plugins/openlibrary/sentry.py), mainly because I thought that we using the `infobase.yml` Sentry configs [here](https://github.com/internetarchive/openlibrary/blob/e52cda783e48cec3ed5b14600a9b27dc89bdcbe6/openlibrary/plugins/openlibrary/sentry.py#L10).

### Proposal & Constraints

Refactor our Sentry code such that it's easy to see, at a glance, which project a `Sentry` instance is associated with.

### PR DESCRIPTION
**Pull request by jimchamp**
Closes #9031

Makes the following Sentry-related changes:
1. Consolidates all Sentry configurations in a single file
2. Creates `SentryFactory` object, which initializes and provides access to each `Sentry` instance

The hope is that this will make our code more readable, and allow us to obtain a reference to any `Sentry` instance, which may be needed to capture notable handled exceptions.

### Technical

A single `SentryFactory` instance is available for import from `/openlibrary/utils`.  The factory exposes `Sentry` instances via the `get_instance` method, which takes a string identifier named `key`.

When called, `get_instance()` checks its `_instances` dict for an instance that maps to the given `key`.  If none are found, `None` is returned.

Before calling `get_instance()`, the `Sentry` instance must have been created using the factory's `create_instance` method.  After the factory is created, an instance is created for each entry in the `sentry.yml` configuration file.

Usage:

```
from openlibrary.utils.sentry import sentry_factory  # imports the exposed `SentryFactory` instance

# Create "ol-web" `Sentry` instance
sentry_factory.create_instance('ol_web', {
    'enabled': True,
    'environment': 'local',
    'dsn': <dsn_url>,
    'traces_sample_rate': 0.01,
  })

# Get instance of Open Library's "ol-web" `Sentry` object
sentry = sentry_factory.get_instance('ol_web')
```

### CODE REVIEW

File: scripts/oldump.py [hunk 1]
```diff
@@ -20,23 +20,23 @@ def log(*args) -> None:
...
         # Squelch output from infobase (needed for sentry setup)
         # So it doesn't end up in our data dumps body
         with open(os.devnull, 'w') as devnull, redirect_stdout(devnull):
-            load_config(ol_config)
-        sentry = Sentry(getattr(config, "sentry_cron_jobs", {}))
-        if sentry.enabled:
-            sentry.init()
-    log(f"sentry.enabled = {bool(ol_config and sentry.enabled)}")
+            # XXX : Does this script require us to load the `OL_CONFIG` for anything other than Sentry?
+            # XXX : If the sentry config is loaded in `sentry.py`, does it need to be loaded here?
```
Comment (jimchamp) [inline scripts/oldump.py:36]: Questions for the reviewer.

File: conf/sentry.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Open Library's Sentry configurations
+
+# Dummy endpoints used for `dns` fields as placeholders.  These must be
```
Comment (jimchamp) [inline conf/sentry.yml:3]:
```suggestion
# Dummy endpoints used for `dsn` fields as placeholders.  These must be
```

File: openlibrary/config.py [hunk 3]
```diff
@@ -28,19 +28,19 @@ def load(config_file):
...
     if 'pytest' in sys.modules:
         # During pytest ensure we're not using like olsystem or something
-        assert config_file == 'conf/openlibrary.yml'
+        assert config_file in ('conf/openlibrary.yml', 'conf/sentry.yml')
     infogami.load_config(config_file)
     setup_infobase_config(config_file)
 
     # This sets web.config.db_parameters
-    server.update_config(config.infobase)
+    server.update_config(config.infobase, overwrite_db_config=overwrite_db_config)
```
Comment (jimchamp) [inline openlibrary/config.py:43]: > [!NOTE]
> This is dependent on [this PR](https://github.com/internetarchive/infogami/pull/218) being merged first. 

We can also not use `load_config` to load the `sentry.yml` configs.

**Review by cdrini — COMMENTED** [review]
This can't be merged in its current state, since it requires a number of changes in olsystem, and a decision on how we want to handle locating the sentry config inside e.g. cron jobs (which currently use the OL_CONFIG environment variable). Is that correct @jimchamp ?

**Comment by jimchamp** [comment]
This PR and the linked, untriaged issue can likely be closed as none of this is necessary.

**Pull request closed without merging**